### PR TITLE
Add Go implementation of POST servers/{id}/queue_update

### LINKED ISSUE #3828: Rewrite /servers/{{id}}/queue_update to Go
## I'm submitting an ...
improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...
- Documentation (possibly)
- Traffic Ops

## Current behavior:
`/servers/{{id}}/queue_update` is handled by Perl

## Expected / new behavior:
`/servers/{{id}}/queue_update` should be handled by Go.

### Minimum API Versions
1.1

### Request Methods
- [ ] `POST`

## Minimal reproduction of the problem with instructions:
N/A

### PR DESCRIPTION
**Pull request by ChrisHines**
## What does this PR (Pull Request) do?
- [x] This PR fixes #3828

## Which Traffic Control components are affected by this PR?
- Traffic Ops

## What is the best way to verify this PR?
Run the added API test against both the Perl and Go backends.

## If this is a bug fix, what versions of Traffic Control are affected?
- master (25181a650cc236028a9f7f9154b5a8919abfa3d0)

## The following criteria are ALL met by this PR
- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by rawlinp** [comment]
add to whitelist

**Comment by ChrisHines** [comment]
> add to whitelist

Is that a request to change something? If so, where is the whitelist and what needs to change?

**Comment by rawlinp** [comment]
That comment was to add your user to the TC asf-ci whitelist so that the PR builder CI will run against your PRs.

File: traffic_ops/traffic_ops_golang/server/queue_update.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusNotFound, nil, nil)
+		return
+	}
+
+	api.WriteResp(w, r, tc.ServerQueueUpdate{
+		ServerID: util.JSONIntStr(serverID),
+		Action:   reqObj.Action,
+	})
+
+	api.CreateChangeLogBuildMsg(
```
Comment (mhoppa) [inline traffic_ops/traffic_ops_golang/server/queue_update.go]: we should catch an error here and return internal server error to the user (prior to writing the response) via the api.HandleErr(....) func

File: traffic_ops/traffic_ops_golang/server/queue_update.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	serverID := int64(inf.IntParams["id"])
+	queue := reqObj.Action == "queue"
+	ok, err := queueUpdate(inf.Tx.Tx, serverID, queue)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, fmt.Errorf("queueing updates: %v", err))
+		return
+	}
+	if !ok {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusNotFound, nil, nil)
```
Comment (mhoppa) [inline traffic_ops/traffic_ops_golang/server/queue_update.go]: nit but I wonder if we include the server id in the user id -> server id {id} not found

File: traffic_ops/traffic_ops_golang/server/queue_update.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		api.Updated,
+		inf.User,
+		inf.Tx.Tx,
+		"server",
+		fmt.Sprint(serverID),
+		map[string]interface{}{"id": serverID},
+	)
+}
+
+func queueUpdate(tx *sql.Tx, serverID int64, queue bool) (bool, error) {
```
Comment (awilliams) [inline traffic_ops/traffic_ops_golang/server/queue_update.go]: Consider a comment describing what the `bool` return value represents.

**Pull request merged**